src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape
+  color: Color
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
```

Comment:
shape should not be availeble in constructor, becasu you can create Circle with Triangle logic

Fix all cases
```suggestion
export class Triangle implements Figure {
  shape = 'triangle'

  constructor(
    public color: Color,
    public a: number,
    public b: number,
    public c: number,
```